cmd/pressure-vessel/main.ino

```diff
@@ -73,19 +73,44 @@ void flash(int n, int p) {
   }
 }
 
+bool pumpOn = false;
+unsigned long startPumpTime = 0;
+int alarmCount = 0; // Global to track alarm state entries.
+
 // This will put arduino into an alarm state i.e. something
 // went wrong.
-void alarmed(int flashes){
-  // Again force relay pin low, we don't want pump on.
+void alarmed(int flashes) {
+  unsigned long alarmStartTime = millis();
+  alarmCount++; // Increment the alarm state counter.
+  
+  // Log that the system has entered the alarm state.
+  Serial.print("Entering alarm state. Count: ");
+  Serial.println(alarmCount);
+
+  // Ensure pump is off for safety.
   relay(LOW);
+  pumpOn = false;
 
-  // Now flash LED to warn of alarm state.
-  while(true){
+  // Stay in alarm state until timeout AND pressure is below MAX_PRESSURE.
+  while (true) {
     flash(flashes, ALARM_PULSE);
-    delay(ALARM_PERIOD - (flashes*ALARM_PULSE));
+    delay(ALARM_PERIOD - (flashes * ALARM_PULSE));
+
+    float currentPressure = read_pressure();
+
+    // Check for timeout AND safe pressure.
+    if ((millis() - alarmStartTime > 30000) && (currentPressure < MAX_PRESSURE)) {
```

Comment:
So we don't do any sort of system restart/reboot ? The major assumption if we don't is that it's a glitch external to the software right ? Shouldn't we do a software reset (https://www.instructables.com/two-ways-to-reset-arduino-in-software/)

Reply:
That's kind of a different feature. Currently, it just acts as a break so the pump doesn't run too long at once, but you're suggesting a full system reset to come out of alarm. I don't think it needs to do that and that wasn't what I was intending so would rather not include it in this PR.

Reply:
We may have diverged in our thinking a bit for this change, but I thought half the point of this was to reset the Nano in case it was some weird state in the software/hardware ? It's OK if you don't want to do it in this change I guess, especially considering you won't be able to test it easily at this point.